### ISSUE TITLE
Auth's resetPassword(), resendSignUpCode() stop to work

### ISSUE THREAD

**Issue by samkudr** [aws-amplify/amplify-swift#2524]
### Describe the bug

Right now in my app I use v1.28.1 and the following code to reset password and to resend confirmation email after signing up:
```
Amplify.Auth.resetPassword(for: email).resultPublisher...
Amplify.Auth.resendSignUpCode(for: email).resultPublisher...
```
Everything works great.

In v2.0.1 these funcs seem to not work. Both of them produce error 'Client 6ii4...p3lq is configured for secret but secret was not received', recovery suggestion—'Check whether the given values are correct and the user is authorized to perform the operation.'

The app uses Cognito User Pool, confirmation method is configured to 'verification link in email'.
Signing in into existing user account (Amplify.Auth.signIn()), the first step of signing up (Amplify.Auth.signUp()) work fine.

### Steps To Reproduce

```swift
1. Switch Amplify package from 1.28 to 2.0.
2. Replace `Amplify.Auth.resetPassword(for: email).resultPublisher` with Combine approach of 2.0 `Amplify.Publisher.create{ try await Amplify.Auth.resendSignUpCode(for: email) }` (async/await approach doesn't work either).
```

### Expected behavior

A reset or confirmation email should be sent to the user.

### Amplify Framework Version

2.0.1

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5

### CLI version

6.3.1

### Xcode version

14.0.1 (14A400)

### Is this a regression?

Yes

### Regression additional context

Upgraded from 1.28.1 to 2.0.1.

### Device

iPad

### iOS Version

iOS 16.1

**Comment by harsh62** [comment]
@samkudr Thank you for raising the issue. Can you share the `amplifyconfigurations.json` redacted all the sensitive information?

**Comment by royjit** [comment]
Thank you for adding the info @samkudr. We are working on a fix right now, will update here when we have more details.

**Comment by royjit** [comment]
Created this PR that should fix the issue https://github.com/aws-amplify/amplify-swift/pull/2528.
Can you please try with the fix branch and see if that fixes your issue? `royjit.addscrethash`

**Comment by samkudr** [comment]
Yes, it fixes the problem. Everything works on iPad and macOS.
Verified from `main`, commit `aa1d240539d5e3f731793eaa1169ca410b8e5253` since branch `royjit.addscrethash` had been already deleted.
That was quick!

**Comment by anush2593** [comment]
Is this issue really closed? I have the same issue.

**Comment by anush2593** [comment]
resendSignUpCode does not work, although other api calls do work

**Comment by atierian** [comment]
Linking to the new issue you (@anush2593) just opened for future reference. https://github.com/aws-amplify/amplify-swift/issues/2791

As mentioned there, the fix for this was released in 2.0.2. Upgrading to that version or higher will resolve the issue you're experiencing.
